cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,492 @@
+	if peErr != nil {
+		return *peErr, nil
+	}
+	deleteFederatedDatabaseAPIRequest := atlas.AtlasV2.DataFederationApi.DeleteFederatedDatabase(context.Background(), *currentModel.ProjectId, *currentModel.TenantName)
+	_, response, err := deleteFederatedDatabaseAPIRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		_, _ = logger.Warnf("Execute error: %s", err.Error())
+		return handleError(response, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Completed",
+		ResourceModel:   nil}, nil
+}
+
+// List handles the List event from the Cloudformation service.
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+	listFederatedDatabaseAPIRequest := atlas.AtlasV2.DataFederationApi.ListFederatedDatabases(context.Background(), *currentModel.ProjectId)
+
+	dataLakeTenants, response, err := listFederatedDatabaseAPIRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		_, _ = logger.Warnf("Execute error: %s", err.Error())
+		return handleError(response, err)
+	}
+	tenants := make([]interface{}, 0)
+	for i := range dataLakeTenants {
+		var tenant Model
+		tenant.getDataLakeTenant(dataLakeTenants[i])
+		tenants = append(tenants, tenant)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List Completed",
+		ResourceModels:  tenants}, nil
+}
+
+func closeResponse(response *http.Response) {
+	if response != nil {
+		response.Body.Close()
+	}
+}
+
+func handleError(response *http.Response, err error) (handler.ProgressEvent, error) {
+	return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error during execution : %s", err.Error()),
+		response), nil
+}
+
+func (model *Model) setDataLakeTenant() (dataLakeTenant atlasSDK.DataLakeTenant, err error) {
+	cloudProvider := constants.AWS
+	if model.DataProcessRegion.CloudProvider != nil {
+		cloudProvider = *model.DataProcessRegion.CloudProvider
+	}
+	dataLakeTenant = atlasSDK.DataLakeTenant{
+		Name: model.TenantName,
+		CloudProviderConfig: &atlasSDK.DataLakeCloudProviderConfig{
+			Aws: atlasSDK.DataLakeAWSCloudProviderConfig{
+				TestS3Bucket: *model.CloudProviderConfig.TestS3Bucket,
+				RoleId:       *model.CloudProviderConfig.RoleId,
+			},
+		},
+		DataProcessRegion: &atlasSDK.DataLakeDataProcessRegion{
+			Region:        *model.DataProcessRegion.Region,
+			CloudProvider: cloudProvider,
+		},
+
+		Storage: model.newDataFederationDataStorage(),
+	}
+	return dataLakeTenant, nil
```

Comment:
the function defines an error but is returning nil every time

Reply:
err not required so removed.